src/modules/AutoinstStorage.rb

```diff
@@ -323,19 +328,72 @@ def preprocessed_settings(settings)
       preprocessor.run(settings)
     end
 
-    # Determine (and warn the user) no storage is available for installation
+    # Errors while importing
+    #
+    # @param partitioning [Array<Hash>] Profile settings (list of drives for custom partitioning)
+    # @return [Array<String>] List of errors
+    def import_errors(partitioning)
+      [unavailable_devices_error(partitioning)].compact
+    end
+
+    # Error when there are no available devices for installing
+    #
+    # @param partitioning [Array<Hash>] Profile settings (list of drives for custom partitioning)
+    # @return [String, nil] Error message; nil when there is no error
+    def unavailable_devices_error(partitioning)
+      return nil if available_devices?(partitioning)
+
+      _("No storage devices were found for the installation.\n" \
+        "Please, check your hardware or your AutoYaST profile.")
+    end
+
+    # Whether there are available devices for installation
+    #
+    # In case that there are no disks, it is still possible to install over NFS.
+    #
+    # @param partitioning [Array<Hash>] Profile settings (list of drives for custom partitioning)
+    # @return [Boolean]
+    def available_devices?(partitioning)
+      available_storage? || nfs_root?(partitioning)
+    end
+
+    # Whether there are storage devices (disks) available for installing
     #
-    # @return [Boolean] true if there are devices for installation; false otherwise.
+    # @return [Boolean]
     def available_storage?
       probed = Y2Storage::StorageManager.instance.probed
       return true if probed && !probed.empty?
-      Yast::Popup.Error(
-        _("No storage devices were found for the installation.\n" \
-          "Please, check your hardware or your AutoYaST profile.")
-      )
+
       false
     end
 
+    # Whether the profile sets root over NFS
+    #
+    # @param partitioning [Array<Hash>] Profile settings (list of drives for custom partitioning)
+    # @return [Boolean]
+    def nfs_root?(partitioning)
+      partitions = nfs_drives(partitioning).map { |d| d["partitions"] }.flatten.compact
```

Comment:
np: What about using `Enumerable#flat_map`?

Reply:
On the one hand, what I do not like about this solution is that this class knows too much about the profile's structure. Extending the `Y2Storage::AutoinstProfile` classes to offer a nicer API to get the NFS drives/partitions would be better. On the other hand, doing so will be more intrusive, as the profile is converted into a proper profile class too late (in the yast2-storage-ng module IIRC). But registering a *fatal* issue (or even launching an exception in the worst case) might be a viable solution.

What do you think?

Reply:
As commented offline, finally we have decided to simply remove the check for unavailable storage. When no devices are available, there is still a possibility to install the system over NFS.

Currently, an issue is always register if the proposal was not possible, so the user would be alerted about "not possible proposal" even when the "guided proposal" is failed.